code/game/objects/items/draftingchalk.dm

```diff
@@ -18,37 +18,77 @@
 			smooth_icon(C)
 
 /obj/item/pen/drafting
-	name = "white drafting chalk"
-	desc = "A piece of white chalk for marking areas of floor."
+	name = "white chalk"
+	desc = "A piece of white chalk for marking areas of floor, or for drawing."
 	icon = 'icons/obj/crayons.dmi'
 	icon_state = "dchalk"
 	color = "#FFFFFF"
 	var/colorName = "whitec"
+	var/linemode = TRUE
+	var/instant = 0
 
 /obj/item/pen/drafting/attack_self(var/mob/user)
+	if(linemode)
+		linemode = FALSE
+		to_chat(user, SPAN_SUBTLE("You will now draw freestyle."))
+	else
+		linemode = TRUE
+		to_chat(user, SPAN_SUBTLE("You will now draw straight lines."))
 	return
 
-/obj/item/pen/drafting/afterattack(turf/target, mob/user, proximity)
-	if (!proximity || !istype(target, /turf/simulated) || target.density || target.is_hole)
-		return
-
-	to_chat(user, "You start marking a line on [target].")
+/obj/item/pen/drafting/examine(mob/user)
+	. = ..()
+	to_chat(user, SPAN_SUBTLE("Use the chalk on itself to change what you will draw."))
 
-	if (!do_after(user, 1 SECONDS, act_target = target))
+/obj/item/pen/drafting/afterattack(turf/target, mob/user, proximity)
+	if(linemode)
+		if (!proximity || !istype(target, /turf/simulated) || target.density || target.is_hole)
+			return
+
+		to_chat(user, "You start marking a line on [target].")
+
+		if(!do_after(user, 1 SECONDS, act_target = target))
+			return
+
+		for (var/obj/effect/decal/cleanable/draftingchalk/C in target)
+			qdel(C)
+
+		to_chat(user, "You mark a line on [target].")
+
+		var/obj/effect/decal/cleanable/draftingchalk/C = new(target)
+		C.color = color
+		target.add_fingerprint(user)
+	else	//code copied graciously from crayon.dm
+		if(!proximity) 
+			return 
+		if(istype(target,/turf/simulated/floor))
```

Comment:
why specifically that type of floor? all we have to check is if it's simulated

Reply:
Blame copypasta from crayons.dm.  I've fixed it.  Maybe there'd be simulated floors the original author didn't want?  shrug

Reply:
Oh, I see @NonQueueingMatt .  It's because without that designation you can draw on walls and stuff. 

...Which I think is actually pretty cool.  So I'll do that?

Reply:
I think that's fine? I'm not sure what that would lead to. Just make sure you can actually clean them off walls

Reply:
Well, I tinkered with it a lot and settled on making it so you can draw non-lines on walls (runes, graffiti, etc.) and only see them from one side, but not lines.  The smooth_icon stuff for the lines was too hard to overcome for me to make it work on walls.  But the ones that I do let be on the walls can be cleaned up by clicking directly on them, or just by cleaning the tile in front of the wall they're on (since that's where the decal is actually located; the thing visible on the wall is just what it's offset to, like a fire extinguisher cabinet.)